### ISSUE TITLE
Some orders create double payment intents

### ISSUE THREAD

**Issue by marcinbot** [Automattic/woocommerce-payments#3998]
### Describe the bug

4860354-zd-woothemes

User reports double-charging certain customers intermittently. We can see the duplicate intents in the Stripe dashboard (with the same order ID), but the order notes only include the first intent.

### To Reproduce

Unclear, need to investigate.

### Expected behavior

The customer is charged only once

### Desktop:

The customer seems to have been using Chrome on Windows.

### Additional context
p1647289194388729-slack-C7U3Y3VMY

**Comment by htdat** [comment]
Asking creds again here p1648723136458479/1647289194.388729-slack-C7U3Y3VMY

**Comment by htdat** [comment]
Based on this convo p1649060633788189/1647289194.388729-slack-C7U3Y3VMY 

Since we're not able to replicate this error and the merchant no longer experiences this issue (at least for now), here is the way forward: 

- **Temporary (applied for sites currently having this issue):** Use this simple plugin to log more info [wcpay-issue-3998.zip](https://github.com/Automattic/woocommerce-payments/files/8416541/wcpay-issue-3998.zip). NOTE: please test on your dev/test site first before activating it on any production site. 
- **Long term (applied for all WCPay sites)**: Add more logging info for this method` WC_Payments_API_Client::request`, similar to the simple plugin above but integrating the code into the current `\WCPay\Logger` usages inside this method.

**Comment by dechov** [comment]
Reopening, as we've seen more occurrences of this issue (paJDYF-4tz-p2), and the logging could be helpful but doesn't constitute a solution

Edit: Adding #5631747-zen as another ticket experiencing the issue.

**Comment by csmcneill** [comment]
5590770-zen
5590770-zen
1420581-hc

**Comment by jessy-p** [comment]
To be observed after https://github.com/Automattic/woocommerce-payments/pull/4854 (4.9.0)

**Comment by jessy-p** [comment]
Closing in favor of https://github.com/Automattic/woocommerce-payments/issues/4858
